genienlp/tasks/almond_task.py

```diff
@@ -455,16 +383,20 @@ def preprocess_field(self, sentence, field_name=None, answer=None, example_id=No
         new_tokens = new_sentence.split(' ')
         new_sentence_length = len(new_tokens)
 
-        tokens_type_ids, tokens_type_probs = None, None
+        tokens_type_ids, tokens_type_probs, token_qids = None, None, None
 
-        if 'type_id' in self.args.ned_features and field_name != 'answer':
+        if self.args.do_ned and 'type_id' in self.args.ned_features and field_name != 'answer':
             tokens_type_ids = [
                 [self.args.ned_features_default_val[0]] * self.args.ned_features_size[0] for _ in range(new_sentence_length)
             ]
-        if 'type_prob' in self.args.ned_features and field_name != 'answer':
+        if self.args.do_ned and 'type_prob' in self.args.ned_features and field_name != 'answer':
             tokens_type_probs = [
                 [self.args.ned_features_default_val[1]] * self.args.ned_features_size[1] for _ in range(new_sentence_length)
             ]
+        if self.args.do_ned and 'qid' in self.args.ned_features and field_name != 'answer':
+            token_qids = [
+                [self.args.ned_features_default_val[2]] * self.args.ned_features_size[2] for _ in range(new_sentence_length)
+            ]
```

Comment:
I just noticed - why is this code in the Almond task and not in the general data loading code? How do we use NED for non-Almond tasks?

Reply:
I'm gradually moving all the code for NED out of almond. Bootleg is now completely task-independent and can be used for other tasks too. The remaining code is for old approaches that use ES/ local database for NER. I will be moving these out also in the next PRs.